packages/flutter/lib/src/animation/tween_sequence.dart

```diff
@@ -0,0 +1,128 @@
+import 'package:flutter/foundation.dart';
+
+import 'animation.dart';
+import 'curves.dart';
+import 'tween.dart';
+
+/// Enables creating an [Animation] whose value is defined by a
+/// sequence of [Tween]s.
+///
+/// Each [TweenSequenceItem] has a weight that defines its percentage
+/// of the animation's duration, and an optional curve that's applied
+/// to the item's tween. Each tween defines the animation's value
+/// during the interval indicated by its weight.
+///
+/// For example, to define an animation that uses an easing curve to
+/// interpolate between 5.0 and 10.0 during the first 40% of the
+/// animation, remain at 10.0 for the next 20%, and then return to
+/// 10.0 for the final 40%:
+///
+/// ```
+/// final Animation<double> = new TweenSequence(
+///   <TweenSequenceItem<double>>[
+///     new TweenSequenceItem<double>(
+///       tween: new Tween<double>(begin: 5.0, end: 10.0),
+///       curve: Curves.ease
+///       weight: 40.0,
+///     ),
+///     new TweenSequenceItem<double>(
+///       tween: new ConstantTween<double>(10.0),
+///       weight: 20.0,
+///     ),
+///     new TweenSequenceItem<double>(
+///       tween: new Tween<double>(begin: 10.0, end: 5.0),
+///       curve: Curves.ease,
+///       weight: 40.0,
+///     ),
+///   ],
+/// ).animate(myAnimationController);
+///```
+class TweenSequence<T> extends Animatable<T> {
```

Comment:
we should mention that actually creating one of these is expensive (so that people don't just create them in build methods and throw them away every frame)